packages/tokenbank_v1/contracts/modules/transfers/QuotaTransfers.sol

```diff
@@ -16,11 +16,257 @@
+        address            to,
+        uint               amount
+        )
+        external
+        nonReentrant
+        onlyFromWalletOwner(wallet)
+        onlyWhenWalletUnlocked(wallet)
+    {
+        if (whitelistStore.isWhitelisted(wallet, to)) {
+            approveInternal(wallet, token, to, amount);
+            return;
+        }
+
+        uint allowance = ERC20(token).allowance(wallet, to);
+        if (allowance >= amount) {
+            approveInternal(wallet, token, to, amount);
+            return;
+        }
+
+        allowance = amount - allowance;
+        uint valueInCNY = priceProvider.getValueInCNY(token, allowance);
+
+        if (quotaStore.checkAndAddToSpent(wallet, valueInCNY)) {
+            approveInternal(wallet, token, to, allowance);
+            return;
+        }
+
+        revert("OUT_OF_QUOTA");
+    }
+
+    function callContract(
+        address            wallet,
+        address            to,
+        uint               amount,
+        bytes     calldata data,
+        bool               enablePending
+        )
+        external
+        nonReentrant
+        onlyWhenWalletUnlocked(wallet)
+        returns (bytes32 pendingTxId)
+    {
+        bytes32 txid = keccak256(
+            abi.encodePacked(
+                "__CALL_CONTRACT__",
+                wallet,
+                to,
+                amount,
+                keccak256(data)
+            )
+        );
+
+        bool foundPendingTx = authorizeWalletOwnerAndPendingTx(wallet, txid);
+        bool allowed = whitelistStore.isWhitelisted(wallet, to);
+        if (!allowed) {
+            uint valueInCNY = priceProvider.getValueInCNY(address(0), amount);
+            allowed = quotaStore.checkAndAddToSpent(wallet, valueInCNY);
+        }
+
+        if (allowed) {
+            callContractInternal(wallet, to, amount, data);
+            if (foundPendingTx) {
+                delete pendingTransactions[wallet][txid];
+            }
+        } else if(!foundPendingTx && enablePending) {
+            createPendingTx(wallet, txid);
+            pendingTxId = txid;
+        }
+    }
+
+    function approveThenCallContract(
+        address            wallet,
+        address            token,
+        address            to,
+        uint               amount,
+        bytes     calldata data
+        )
+        external
+        nonReentrant
+        onlyFromWalletOwner(wallet)
+        onlyWhenWalletUnlocked(wallet)
+        notWalletOrItsModule(wallet, to)
+    {
+        if (whitelistStore.isWhitelisted(wallet, to)) {
+            approveInternal(wallet, token, to, amount);
+            callContractInternal(wallet, to, 0, data);
+            return;
+        }
+
+        uint allowance = ERC20(token).allowance(wallet, to);
+        if (allowance >= amount) {
+            approveInternal(wallet, token, to, amount);
+            callContractInternal(wallet, to, 0, data);
+            return;
+        }
+
+        allowance -= amount;
```

Comment:
Still needs to be changed to `allowance = amount - allowance;` like above.

Reply:
fixed